### PR TITLE
Bussines logic for unPythonicGetterSetterViolation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: wemake_python_styleguide/logic/tree/classes.py [hunk 1]
```diff
@@ -34,3 +39,54 @@ def is_forbidden_super_class(class_name: Optional[str]) -> bool:
...
+
+
+def get_attributes(
+    node: ast.ClassDef,
+) -> Tuple[List[types.AnyAssign], List[ast.Attribute]]:
+    """Returns all class and instance attributes of a class."""
+    class_attributes = []
+    instance_attributes = []
+
+    for nd in ast.walk(node):
```
Comment (adrwes) [inline wemake_python_styleguide/logic/tree/classes.py:51]: Maybe nd -> subnode for consistency

File: wemake_python_styleguide/logic/tree/classes.py [hunk 2]
```diff
@@ -34,3 +39,54 @@ def is_forbidden_super_class(class_name: Optional[str]) -> bool:
...
+        if isinstance(subnode, FunctionNodes) and correct_context:
+            if is_getter_or_setter(subnode):
+                method_postfixes.add(subnode.name.partition('get_')[2])
+                method_postfixes.add(subnode.name.partition('set_')[2])
+    return method_postfixes
+
+
+def is_getter_or_setter(node: types.AnyFunctionDef) -> bool:
+    """Checks if non property decorated function contains get or set prefix."""
+    if any(prefix in node.name for prefix in ('get_', 'set_')):
```
Comment (adrwes) [inline wemake_python_styleguide/logic/tree/classes.py]: Would this give the correct behavior for x_get_x or x_set_x?

**Review by adrwes — APPROVED** [review]
Otherwise LGTM!

File: wemake_python_styleguide/visitors/ast/classes.py [hunk 3]
```diff
@@ -117,6 +120,24 @@ def _is_correct_base_class(self, base_class: ast.AST) -> bool:
...
+    def _check_getters_setters_methods(self, node: ast.ClassDef) -> None:
+        class_attributes, instance_attributes = classes.get_attributes(node)
+        attribute_names = set(
+            name_nodes.flat_variable_names(class_attributes),
+        ).union({
+            instance.attr for instance in instance_attributes
+        })
+
+        for attribute in attribute_names:
+            for method_postfix in classes.getter_setter_postfixes(node):
```
Comment (adrwes) [inline wemake_python_styleguide/visitors/ast/classes.py:131]: Would this give the correct behavior for an attribute blabla_x? Maybe EndsWith should be used?

Comment (adrwes) [reply wemake_python_styleguide/visitors/ast/classes.py:131]: Sorry I misread the code.

File: wemake_python_styleguide/logic/tree/classes.py (hunk 1 above)
Comment (ruwaid4) [reply wemake_python_styleguide/logic/tree/classes.py]: Will fix

Comment (ghost) [reply wemake_python_styleguide/logic/tree/classes.py:51]: It seems to have been shortened to pass the linters line too long violation.

File: wemake_python_styleguide/logic/tree/classes.py (hunk 2 above)
Comment (ghost) [reply wemake_python_styleguide/logic/tree/classes.py]: Will be fixed

**Pull request merged**